avocado/core/result.py

```diff
@@ -165,6 +166,7 @@ def end_test(self, state):
         """
         self.tests_run += 1
```

Comment:
as you're storing the full state, you should remove the `self.tests_run` as the information is duplicate.